Port trace in xbar when address error

yanleeG opened the pull request:
When xbar encounters the address error, print out the port trace in the packet for user to debug if the port trace is enabled.

To gain the packet of the access, the parameter of findPort() function is changed from AddrRange to PacketPtr.

When running gem5 with "--debug-flags=PortTrace", we can see the full path of the unexpected access when xbar cannot find the destination of the address.

powerjg commented:
Overall, this change is good with me. However, it looks like there are some other places that use the `findPort` function: https://github.com/gem5/gem5/actions/runs/5854325967/job/15870511333?pr=180#step:5:4255

yanleeG committed:
mem: dump out port trace when address decode error

1. Add findPort(PacketPtr pkt) for getting the port trace from the Packet.
   Keep the findPort(AddrRange addr_range) for recvMemBackdoorReq(...)
2. With the debug flag `PortTrace` enabled, user can see the full path of
   the packet with the corresponding address when address error in xbar.

Change-Id: Iaf43ee2d7f8c46b9b84b2bc421a6bc3b02e01b3e

yanleeG committed:
mem: port: add address value in the port trace

Add the address value from the packet with the request port name.

Change-Id: I3d4c75f48ca6fbdbd5656e594d5f85f9e5626be8

yanleeG committed:
mem: port: add getTraceInString() method

Return the whole port trace of the packet as a string.

Change-Id: I7b1b1fef628a47a6ce147cb5fb75af81948c1d89

yanleeG commented:
> Overall, this change is good with me. However, it looks like there are some other places that use the `findPort` function: https://github.com/gem5/gem5/actions/runs/5854325967/job/15870511333?pr=180#step:5:4255

Thanks for pointing this out! I did not notice that my commit does not cover all findPort() function. In order to support the original `findPort(AddrRange)` within the `CoherentXBar::recvMemBackdoorReq`, I make the packet pointer optional in `findPort()`.

powerjg reviewed (approved):
Great! I like that this is fully "backwards compatible" as well :)
